integration/src/test/java/org/mule/test/routing/SynchronizedFlowRefOnMuleContextStartTestCase.java

```diff
@@ -82,50 +73,18 @@ private void prePopulateObjectStore() throws ObjectStoreException {
     ObjectStore<Event> objectStore = muleContext.getRegistry().lookupObject("objectStore");
 
     Message testMessage = Message.builder().payload(TEST_MESSAGE).build();
-    Flow clientFlow = muleContext.getRegistry().get("flow2");
+    Flow clientFlow = muleContext.getRegistry().get(FLOW_UNDER_TESTING);
     Event testMuleEvent =
         Event.builder(create(clientFlow, fromSingleComponent(clientFlow.getName()))).message(testMessage).build();
     objectStore.store(testMuleEvent.getCorrelationId(), testMuleEvent);
   }
 
-  public static class UnblockProcessingSource implements MessageSource, Startable {
-
-    @Override
-    public void start() throws MuleException {
-      waitMessageInProgress.release();
-    }
-
-    @Override
-    public Object getAnnotation(QName name) {
-      return null;
-    }
-
-    @Override
-    public Map<QName, Object> getAnnotations() {
-      return null;
-    }
-
-    @Override
-    public void setAnnotations(Map<QName, Object> annotations) {
-
-    }
-
-    @Override
-    public ComponentLocation getLocation() {
-      return null;
-    }
-
-    @Override
-    public void setListener(Processor listener) {
-
-    }
-  }
-
   public static class TestMessageProcessorCallback implements EventCallback {
 
     @Override
     public void eventReceived(Event event, Object component, MuleContext muleContext) throws Exception {
-      if (waitMessageInProgress.await(0, MILLISECONDS)) {
+      if (((SkeletonSource) ((AbstractPipeline) muleContext.getRegistry().lookupFlowConstruct(FLOW_UNDER_TESTING)).getSource())
```

Comment:
Could not assert here directly because i think this callback is executed in a different thread than the test call itself (waitsForStartedMuleContextBeforeAttemptingToSendMessageToEndpoint()). Since the test thread would have finished before the assert done here, it would've never failed.